src/item.cpp

```diff
@@ -3965,6 +3984,19 @@ bool item::ammo_consume( int qty ) {
     return false;
 }
 
+itype_id item::ammo_current() const
+{
+    if( is_ammo() ) {
+        return typeId();
```

Comment:
> Incidentally we use  "NULL"  and  "null"  throughout the codebase.

That's why I came up with `string_id<T>::NULL_ID` (see string_id.h). It is, however, of the type `string_id<T>`, not just a plain `std::string`. Using it avoids having to remember what the null id actually is.

For example: it's "mon_null" for monster type, but that string appears exactly once as string literal in the source, all other places use `NULL_ID`.

> Would there be any objection to a separate PR that converts everything to the same and if so which is the preferred form?

I would avoid that, to much work. You'll have to change various JSON files or translate the old null-ids to new null-ids.

> Is  type->id  ever an empty string?

`type->id` is always a valid item type id, so yes, this should work fine.